spec/services/research_safety_adapters/scishield_training_synchronizer_spec.rb

```diff
@@ -9,66 +9,88 @@
   let(:satus_code) { nil }
   let(:api_endpoint) { ResearchSafetyAdapters::ScishieldApiClient.new.api_endpoint(user.email) }
 
-  before do
-    stub_request(:get, api_endpoint)
-      .to_return(
-        status: status_code,
-        body: File.new(response)
-      )
+  describe "#retry_max" do
+    it "returns an integer" do
+      expect(synchronizer.retry_max).to be_a(Integer)
+    end
   end
 
-  context "when the API responds without error" do
-    let(:status_code) { "200" }
+  describe "#batch_size" do
+    it "returns an integer" do
+      expect(synchronizer.batch_size).to be_a(Integer)
+    end
+  end
 
-    it "adds courses to database" do
-      expect(ScishieldTraining.count).to eq 0
-      synchronizer.synchronize
-      expect(ScishieldTraining.count).to eq 3
-      expect(ScishieldTraining.all.map(&:course_name)).to contain_exactly(*course_names)
+  describe "#batch_sleep_time" do
+    it "returns an integer" do
+      expect(synchronizer.batch_sleep_time).to be_a(Integer)
     end
   end
 
-  context "when the API responds with error" do
-    before { 2.times { create(:scishield_training, user_id: 1) } }
+  context "API requests" do
+    before do
+      stub_request(:get, api_endpoint)
+        .to_return(
+          status: status_code,
+          body: File.new(response)
+        )
+    end
 
-    context "when the response is missing the data attribute" do
-      let(:response) { File.expand_path("../../fixtures/scishield/empty_success.json", __dir__) }
+    context "when the API responds without error" do
       let(:status_code) { "200" }
 
-      it "does not add courses to database" do
-        expect(ScishieldTraining.count).to eq 2
-        synchronizer.synchronize
-        expect(ScishieldTraining.count).to eq 2
+      it "adds courses to database" do
+        expect(ScishieldTraining.count).to eq 0
+        # Note, this will take take 20 seconds to run, to change this, update
+        # the value of `synchronizer.batch_sleep_time`
```

Comment:
Can we change this value for this spec?  I think you can use metadata for `feature_setting` here, right?